src/models/onnxruntime_api.h

```diff
@@ -108,49 +198,43 @@ inline void InitApi() {
   //
 
   const std::string path = "libonnxruntime.so";  // "libonnxruntime4j_jni.so" is also an option if we have issues
-  __android_log_print(ANDROID_LOG_INFO, "GenAI", "Attempting to dlopen %s native library", path.c_str());
+  void* ort_lib_handle = LoadDynamicLibraryIfExists(path);
 
-  using OrtApiBaseFn = const OrtApiBase* (*)(void);
-  OrtApiBaseFn ort_api_base_fn = nullptr;
-
-  void* ort_lib_handle = dlopen(path.c_str(), RTLD_LOCAL);
+#if !defined(__ANDROID__)
+  // Search for pip installation
   if (ort_lib_handle == nullptr) {
-    __android_log_assert("ort_lib_handle != nullptr", "GenAI", "Failed to load %s", path.c_str());
-  }
-
-  ort_api_base_fn = (OrtApiBaseFn)dlsym(ort_lib_handle, "OrtGetApiBase");
-  if (ort_api_base_fn == nullptr) {
-    __android_log_assert("ort_api_base_fn != nullptr", "GenAI", "OrtGetApiBase not found");
+    const std::array<std::string, 4> target_libraries = {
+        std::string("libonnxruntime.so"),
+        std::string("libonnxruntime.so.1.18.0"),
+        std::string("libonnxruntime.so.1.19.0"),
+        std::string("libonnxruntime.so.1.20.0")};
```

Comment:
To reviewer: the ORT lib bundled in ort-nightly package has version number. This seems to be the only way to have forward compatibility. I don't like it, but I don't have better idea.

Reply:
The SONAME is now `libonnxruntime.so.1`, but the file name is still `libonnxruntime.so.1.19.0`.